What is Word2Vec (#1730)
section-engineering-education/engineering-education: articles/what-is-word2vec/index.md

#### Training

The system is trained using backpropagation and stochastic gradient descent to tune the weights and reduce the cost function. By doing so, those weights become the vectors for the words in question.

### Key takeaways

1. The Continuous Bag-of-Words (CBOW) model uses context words to predict the target words. Conversely, the skip-gram model, uses the target words to predict the context words.
2. The Word2Vec algorithm causes words that have similar contexts to have similar vector embeddings (be closer together) and words that have different contexts to be far apart.
3. Skipgram is relatively slower than CBOW and usually works well with a large corpus of training data.
4. CBOW is faster to train than skip-gram. 

### Wrapping Up

I hope this article has been helpful to you. Word2vec was an extremely influential algorithm in NLP until recently when it went out of fashion as far as research is concerned primarily with the rise of language models such as Google's [BERT](https://en.wikipedia.org/wiki/BERT_(language_model)) and [ELMo](https://allennlp.org/elmo). However, it is still very relevant.

> It is also important to note that word2vec isn't the only word embedding models that work well. Stanford's [GloVe](https://en.wikipedia.org/wiki/GloVe_(machine_learning)) and Facebook's [FastText](https://fasttext.cc/) are examples of other popular word embedding models you can explore that also works so great. Feel free to also read on it.

Reviewer: ```suggestion
It is also important to note that word2vec isn't the only word embedding model that works well. Stanford's [GloVe](https://en.wikipedia.org/wiki/GloVe_(machine_learning)) and Facebook's [FastText](https://fasttext.cc/) are examples of other popular word embedding models you can explore that also work well. Feel free to also have a read on them.
```
